common/oauthTokenManager.go

```diff
@@ -771,13 +770,41 @@ func (credInfo *OAuthTokenInfo) GetTokenCredential() (azcore.TokenCredential, er
 	}
 }
 
+// MarshalJSON customizes the JSON encoding for AutoLoginTypeJSON
+func (alt AutoLoginTypeJSON) MarshalJSON() ([]byte, error) {
+	return json.Marshal(alt.AutoLoginType)
```

Comment:
Let's do the .string call here instead, given we'd prefer to support it as strings

Reply:
done